index.js

```diff
@@ -452,6 +686,9 @@ Command.prototype.allowUnknownOption = function(arg) {
  */
 
 Command.prototype.parse = function(argv) {
+  // trigger autocomplete first
+  this.autocomplete(argv);
```

Comment:
parse will always call autocomplete.

For normal cases, overhead should be acceptable since we only have several string comparison operations. 

If argv shows that this request is likely for autocomplete, omelette will take over and program.exit(0) after processing.

Reply:
Maybe we should better hide auto-completion feature behind the flag and set it to `false` by default?